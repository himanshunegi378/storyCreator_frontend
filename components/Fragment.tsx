import React from 'react'

export interface IFragment {
  like: number
  text: string
}

export interface FragmentProps extends IFragment {}

function Fragment(props: FragmentProps) {
  const { like, text } = props
  return (
    <div className="px-4 py-2 bg-blue-50">
      <div data-testid="text">{text}</div>
      <div data-testid="like"> {like} </div>
    </div>
  )
}

export default Fragment
