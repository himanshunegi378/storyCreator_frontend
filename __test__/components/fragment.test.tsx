import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Fragment, { IFragment } from '../../components/Fragment'

afterEach(cleanup)

describe('Fragment', () => {
  it('match snapshot', () => {
    const fragmentProps: IFragment = { like: 10, text: 'himanshu' }
    const { asFragment } = render(<Fragment {...fragmentProps} />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should show provided text', () => {
    const fragmentProps: IFragment = { like: 10, text: 'himanshu' }
    const { getByTestId } = render(<Fragment {...fragmentProps} />)
    expect(getByTestId('text')).toHaveTextContent('himanshu')
  })

  it('should show provided likes', () => {
    const fragmentProps: IFragment = { like: 10, text: 'himanshu' }
    const { getByTestId } = render(<Fragment {...fragmentProps} />)
    expect(getByTestId('like')).toHaveTextContent('10')
  })
})
