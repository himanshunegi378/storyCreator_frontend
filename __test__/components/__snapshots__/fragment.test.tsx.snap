// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment match snapshot 1`] = `
<DocumentFragment>
  <div
    class="px-4 py-2 bg-blue-50"
  >
    <div
      data-testid="text"
    >
      himanshu
    </div>
    <div
      data-testid="like"
    >
       10 
    </div>
  </div>
</DocumentFragment>
`;
